core/src/main/scala/frameless/InjectionEnum.scala

```diff
@@ -0,0 +1,54 @@
+package frameless
+
+import scala.reflect.{ClassTag, classTag}
+import shapeless._
+
+object InjectionEnum {
+  type InjectionEnum[A] = Injection[A, String]
+
+  def apply[A: InjectionEnum]: InjectionEnum[A] = implicitly[InjectionEnum[A]]
+
+  def instance[A](f: A => String, g: String => A): InjectionEnum[A] =
```

Comment:
Why not use `Injection` directly? Is it for type inference? If you decide to keep it please make it private!